### PR TITLE
Pr nicolen component base class

### PR DESCRIPTION
**Pull request by razstvien01**
PR for the FF:

Toggled in Component List:

1. Added Sample Component Button
2. Mouse Move/Hover: for the Guide Shape of the Component
3. The component always at the front of the layers
4. The component's relative position to the other page fix.

### CODE REVIEW

File: App/scripts/index.js [hunk 1]
```diff
@@ -60,4 +139,13 @@ function toggleComponentList()
     }
 }
 
-
+//* Event Button Functions based on their button name
+function handleComponentButtonClick(buttonName)
+{
+    if(buttonName == "0")
+    {
+        log('Circle button clicked');
```
Comment (RevyaS) [inline App/scripts/index.js]: Remove console logs

File: App/scripts/shapes/Page.js [hunk 2]
```diff
@@ -1,56 +1,319 @@
...
+    showCircleGuideLine(mousePos)
+    {
+        //* Compute Relative Position of mouse to Page        
+        this.circleGuideLine.setXYPosition(mousePos.x, mousePos.y);
+        this.circleGuideLine.setOpacity(1);
+        
+        let maxZIndex = this.group.children.length - 1;
+        this.circleGuideLine.group.zIndex(maxZIndex);
+        
+        let log = console.log;
```
Comment (RevyaS) [inline App/scripts/shapes/Page.js]: Remove console logs

File: App/scripts/shapes/Page.js [hunk 3]
```diff
@@ -1,56 +1,319 @@
...
+        
+        log('MaxZIndex ' + maxZIndex);
```
Comment (RevyaS) [inline App/scripts/shapes/Page.js]: Remove console logs

File: App/scripts/shapes/Page.js [hunk 4]
```diff
@@ -1,56 +1,319 @@
...
+        log(mousePos);
```
Comment (RevyaS) [inline App/scripts/shapes/Page.js]: Remove console logs

**Review by RevyaS — CHANGES_REQUESTED** [review]
Added Comments, please resolve

File: App/scripts/index.js (hunk 1 above)
Comment (razstvien01) [reply App/scripts/index.js]: done

File: App/scripts/shapes/Page.js (hunk 3 above)
Comment (razstvien01) [reply App/scripts/shapes/Page.js]: done

File: App/scripts/shapes/Page.js (hunk 4 above)
Comment (razstvien01) [reply App/scripts/shapes/Page.js]: done

**Pull request closed without merging**